Sources/SmithyXML/Writer/Writer+libxml2.swift

```diff
@@ -13,7 +13,7 @@ extension Writer {
 
     /// Translates this Writer and its children into XML ready to be sent.
     /// - Returns: A `Data` value containing this writer's UTF-8 XML representation.
-    func xmlString() -> Data {
+    public func data() throws -> Data {
```

Comment:
Conforms to `SmithyWriter` and modified to ignore nodes that do not have any XML content; this removes the need to call `detach()` on unused Writers.